src/app/core/store/documents/documents.effects.ts

```diff
@@ -155,4 +178,63 @@ export class DocumentsEffects {
               private store$: Store<AppState>) {
   }
 
+  private createAddCollectionAttributesAction(collection: CollectionModel, document: DocumentModel): Action {
+    const collectionCopy = {...collection, documentsCount: collection.documentsCount + 1};
+
+    const addedAttributes = Object.keys(document.data);
+
+    this.incrementOrAddAttributesInCollection(collectionCopy, addedAttributes);
+
+    return new CollectionsAction.UpdateSuccess({collection: collectionCopy});
+  }
+
+  private createUpdateCollectionAttributesAction(collection: CollectionModel, document: DocumentModel, oldDocument: DocumentModel): Action {
+    const collectionCopy = {...collection};
+
+    const oldAttributes = Object.keys(oldDocument.data);
+    const newAttributes = Object.keys(document.data);
+
+    const addedAttributes = newAttributes.filter(attr => !oldAttributes.includes(attr));
+    const removedAttributes = oldAttributes.filter(attr => !newAttributes.includes(attr));
+
+    this.incrementOrAddAttributesInCollection(collectionCopy, addedAttributes);
+    this.decrementAttributesInCollection(collectionCopy, removedAttributes);
+
+    return new CollectionsAction.UpdateSuccess({collection: collectionCopy});
+  }
+
+  private createRemoveCollectionsAttributesAction(collection: CollectionModel, document: DocumentModel): Action {
+    const collectionCopy = {...collection, documentsCount: Math.max(collection.documentsCount - 1, 0)};
+
+    const deletedAttributes = Object.keys(document.data);
+    this.decrementAttributesInCollection(collectionCopy, deletedAttributes);
```

Comment:
Not important, just code style. Much cleaner solution would be to return a list of attributes from the function and not modify its parameters.